// vuestic-ui/dist/esm-node/src/components/va-list/VaListLabel.mjs
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
import { defineComponent, computed, openBlock, createElementBlock, normalizeStyle, renderSlot } from "vue";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.mjs";

import { u as useColors } from "../../composables/useColors.mjs";


import { _ as _export_sfc } from "../../../plugin-vue_export-helper.mjs";
var VaListLabel_vue_vue_type_style_index_0_lang = "";
const _sfc_main = defineComponent({
  name: "VaListLabel",
  props: __spreadProps(__spreadValues({}, useComponentPresetProp), {
    color: { type: String, default: "primary" }
  }),
  setup: (props) => {
    const { getColor } = useColors();
    return {
      computedStyle: computed(() => ({
        color: getColor(props.color)
      }))
    };
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", {
    class: "va-list-label",
    style: normalizeStyle(_ctx.computedStyle)
  }, [
    renderSlot(_ctx.$slots, "default")
  ], 4);
}
var _VaListLabel = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaListLabel as _ };
